### PR TITLE
rekey stake_redelegate_instruction feature

### PR DESCRIPTION
**Pull request by HaoranYi**
#### Problem

https://github.com/solana-labs/solana/issues/26765

devnet has this feature activated already without the fixes from #32606
unfortunately. Fortunately, testnet it's now having this feature disabled
thanks to the 1.14/1.16 restart. 

In order to turn off this feature from devnet during the next devnent upgrade,
we will need to rekey this feature. 

#### Summary of Changes

Rekey stake_redelegate_instruction feature

Fixes #

### CODE REVIEW

**Review by joncinque — COMMENTED** [review]
Looks good to me, but you'll need Michael's approval since it's his key

**Review by mvines — APPROVED** [review]
ty

**Comment by mvines** [comment]
please of course update the feature-gate issue with the new key information once this PR lands and is backported to v1.16

**Comment by HaoranYi** [comment]
> please of course update the feature-gate issue with the new key information once this PR lands and is backported to v1.16

yeah. updated.

**Pull request merged**
